src/nupic/encoders/MultiEncoder.hpp

```diff
@@ -0,0 +1,202 @@
+/* ---------------------------------------------------------------------
+* Numenta Platform for Intelligent Computing (NuPIC)
+* Copyright (C) 2014-2016, Numenta, In  Unless you have an agreement
+* with Numenta, In, for a separate license for this software code, the
+* following terms and conditions apply:
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Affero Public License version 3 as
+* published by the Free Software Foundation.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+* See the GNU Affero Public License for more details.
+*
+* You should have received a copy of the GNU Affero Public License
+* along with this program.  If not, see http://www.gnu.org/licenses.
+*
+* http://numenta.org/licenses/
+* ---------------------------------------------------------------------
+*/
+
+#ifndef NTA_multiencoder_HPP
+#define NTA_multiencoder_HPP
+
+#include <iostream>
+#include <string> 
+#include <vector>
+#include <tuple>
+#include <map> 
+
+#include "Encoder.hpp"       
+#include "ScalarEncoder2.hpp"    
+#include "AdaptiveScalarEncoder.hpp"    
+#include "DateEncoder.hpp"    
+#include "CategoryEncoder.hpp"    
+#include "SDRCategoryEncoder.hpp"    
+#include "PassThroughEncoder.hpp"    
+#include "CoordinateEncoder.hpp"    
+#include "GeospatialCoordinateEncoder.hpp"    
+#include "RandomDistributedScalarEncoder.hpp"   
+
+#include "Utils.hpp" 
+#include "RangeList.hpp"
+#include "DecodeResult.hpp"
+
+#include <boost/any.hpp>
+
+using namespace std;
+
+namespace encoders {
+
+	/**
+	* A MultiEncoder encodes a dictionary or object with
+	* multiple components. A MultiEncode contains a number
+	* of sub-encoders, each of which encodes a separate component.
```

Comment:
`SDR_Concatenate` can do the same thing as the `MultiEncoder`

Reply:
even with decode? 
What if we keep multiencoder and implement it with SDR_concatenate to keep the convenience calls?

Reply:
`SDR_Concatenate` can not do decode, but it could be extended to do decode as follows:
 * Change `SDR_Concatenate` to work in reverse, so an assignment to the concatenation would split up & write the data into the input SDRs.
 * Call decode on the input SDRs and their encoders.

Reply:
we need #300 for this, the MultiEnc can be removed, or just simplified its implementation